app/controllers/api/red_hat_cloud_service_providers_controller.rb

```diff
@@ -0,0 +1,40 @@
+module Api
+  class RedHatCloudServiceProvidersController < BaseController
+    def red_hat_cloud_service_providers_search_conditions
+      { :type => provider_types }
+    end
+
+    def find_red_hat_cloud_service_providers(id)
+      collection_type = :red_hat_cloud_service_providers
+      klass = collection_class(collection_type)
+      provider = find_resource(klass, "id", id)
+      provider = filter_resource(provider, collection_type, klass)
+      raise NotFoundError, "Invalid #{provider_title} id:#{id} specified" unless provider_types.include?(provider.type)
+
+      provider
+    end
+
+    def sync_resource(_type, id, _data)
+      provider = find_red_hat_cloud_service_providers(id)
+      desc = "Syncing #{provider_ident(provider)}"
+      task_id = Cfme::CloudServices::InventorySync.sync_queue(User.current_user.userid, provider)
+      action_result(true, desc, :task_id => task_id)
+    rescue => e
+      action_result(false, e.to_s)
+    end
+
+    private
+
+    def provider_ident(provider)
+      "#{provider_title} id:#{provider.id} name:'#{provider.name}'"
+    end
+
+    def provider_title
+      "Red Hat Cloud Services Provider"
+    end
+
+    def provider_types
+      Cfme::CloudServices::ManifestFetcher.fetch["manifest"].keys.reject { |k| k == "core" }.uniq
+    end
```

Comment:
I feel like this method should live in `Cfme::CloudServices::ManifestFetcher` itself.  Otherwise the knowledge of "core" is being duplicated in cases, and if we add a new type in the future, we have to remember to also change it here.

Reply:
Or perhaps in the one of the other classes that make up the chain...not sure exactly

BTW, this can be for a follow up PR.

Reply:
Eee so this is going to download the manifest to get the list of providers, then later download the manifest again for collection

Reply:
yeah, unfortunately...I told @abellotti just start with this, then we can make it different later.

Reply:
Yeah, we could do something like cache_with_timeout in the ManifestFetcher (in a follow-up) until we have the ability to download and store on the filesystem the manifest file

Reply:
caching sounds about right here 👍

Reply:
https://github.com/RedHatCloudForms/cfme-cloud_services/pull/25 Should cover it

Reply:
Merged